auto_submit/lib/request_handling/authentication.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+import 'package:shelf/shelf.dart';
+
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../model/google/token_info.dart';
+import '../requests/exceptions.dart';
+import '../service/config.dart';
+import '../service/log.dart';
+
+/// Class capable of authenticating [HttpRequest]s.
+///
+/// There are two types of authentication this class supports:
+///
+///  1. If the request has the `'X-Appengine-Cron'` HTTP header set to "true",
+///     then the request will be authenticated as an App Engine cron job.
+///
+///     The `'X-Appengine-Cron'` HTTP header is set automatically by App Engine
+///     and will be automatically stripped from the request by the App Engine
+///     runtime if the request originated from anything other than a cron job.
+///     Thus, the header is safe to trust as an authentication indicator.
+///
+///  2. If the request has the `'X-Flutter-IdToken'` HTTP header
```

Comment:
This won't be supported for auto submit

Reply:
Hi Casey, the previous logic check two headers: 'X-Appengine-Cron' and 'X-Flutter-IdToken'.
Do you mean that here for autosubmit, we only need to check 'X-Appengine-Cron' and we don't need to take care about the 'X-Flutter-IdToken'?

Reply:
Correct. X-Flutter-IdToken is for Google accounts

Reply:
Gotcha Casey. Since we don't need to check for the Google accounts, I also don't need to get the OAuth client Id. I will also delete all logics for the OAuth client Id in Config.